homeassistant/components/energyid/energyid_sensor_mapping_flow.py

```diff
@@ -0,0 +1,156 @@
+from homeassistant.helpers import entity_registry as er
+from homeassistant.helpers.selector import EntitySelector, EntitySelectorConfig
+
+from .const import CONF_ENERGYID_KEY, CONF_HA_ENTITY_ID, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+# --- Start of Helper Functions ---
+# These functions are now included directly in the file.
+
+
+@callback
+def _get_suggested_entities(hass: HomeAssistant) -> list[str]:
+    """Return a sorted list of suggested sensor entity IDs for mapping."""
+    _LOGGER.debug("Starting _get_suggested_entities")
+    ent_reg = er.async_get(hass)
+
+    suitable_entities = []
+    for entity_entry in ent_reg.entities.values():
+        _LOGGER.debug("Evaluating entity: %s", entity_entry.entity_id)
+        if not (
+            entity_entry.domain == Platform.SENSOR and entity_entry.platform != DOMAIN
+        ):
+            _LOGGER.debug(
+                "Skipping entity %s due to domain/platform checks",
+                entity_entry.entity_id,
+            )
+            continue
+
+        state_class = (entity_entry.capabilities or {}).get("state_class")
+        is_likely_numeric = (
+            state_class
+            in (
+                SensorStateClass.MEASUREMENT,
+                SensorStateClass.TOTAL,
+                SensorStateClass.TOTAL_INCREASING,
+            )
+            or entity_entry.device_class
+            in (
+                SensorDeviceClass.ENERGY,
+                SensorDeviceClass.GAS,
+                SensorDeviceClass.POWER,
+                SensorDeviceClass.TEMPERATURE,
+                SensorDeviceClass.VOLUME,
+            )
+            or entity_entry.original_device_class
+            in (
+                SensorDeviceClass.ENERGY,
+                SensorDeviceClass.GAS,
+                SensorDeviceClass.POWER,
+                SensorDeviceClass.TEMPERATURE,
+                SensorDeviceClass.VOLUME,
+            )
+        )
+        current_state = hass.states.get(entity_entry.entity_id)
+        if current_state and current_state.state not in (
+            STATE_UNKNOWN,
+            STATE_UNAVAILABLE,
+        ):
+            try:
+                float(current_state.state)
+            except (ValueError, TypeError):
+                _LOGGER.debug(
+                    "Entity %s state cannot be converted to float",
+                    entity_entry.entity_id,
+                )
+                continue
+            suitable_entities.append(entity_entry.entity_id)
+            _LOGGER.debug(
+                "Added entity %s to suitable entities", entity_entry.entity_id
+            )
+        elif (
+            is_likely_numeric
+            and current_state
+            and current_state.state != STATE_UNAVAILABLE
+        ):
+            suitable_entities.append(entity_entry.entity_id)
+            _LOGGER.debug(
+                "Added likely numeric entity %s to suitable entities",
+                entity_entry.entity_id,
+            )
```

Comment:
I don't think these checks are needed, the sensor base class raises if sensors have non numeric values since 2 years: https://github.com/home-assistant/core/blob/815b26dcac4668be1bb32eb65730e5718e291bbd/homeassistant/components/sensor/__init__.py#L682-L687